src/ast/mod.rs

```diff
@@ -5593,6 +5609,21 @@ impl fmt::Display for FunctionArgumentClause {
     }
 }
 
+/// A method call
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub struct Method {
+    pub expr: Box<Expr>,
+    pub method: Function,
+}
```

Comment:
representation wise I'm thinking we could use something like
```rust
struct Method {
    pub prefix: Box<Expr>,
    pub suffix: Vec<Function>
}
```
Actually I'm not super sure naming wise of the properties, maybe we can come up with something better. But the main point/change I'm thinking would be the overall representation. Instead of using a nested datastructure for this, that doesn't seem required, the syntax is just a chain of expression so we can just use list to store that chain - and it maps nicely to how the parsing is done since its clear that the first part can be any expr, while subsequent components must be functions. wdyt?

Reply:
good advice!